filebeat/tests/system/test_autodiscover.py

```diff
@@ -64,41 +44,47 @@ def test_default_settings(self):
         """
         Test docker autodiscover default config settings
         """
-        import docker
-        docker_client = docker.from_env()
-
-        self.render_config_template(
-            inputs=False,
-            autodiscover={
-                'docker': {
-                    'cleanup_timeout': '0s',
-                    'hints.enabled': 'true',
-                    'hints.default_config': '''
-                      type: log
-                      paths:
-                        - %s/${data.container.image}.log
-                    ''' % self.working_dir,
+        with self.container_running() as container:
+            self.render_config_template(
+                inputs=False,
+                autodiscover={
+                    'docker': {
+                        'cleanup_timeout': '0s',
+                        'hints.enabled': 'true',
+                        'hints.default_config': '''
+                          type: log
+                          paths:
+                            - %s/${data.container.name}.log
+                        ''' % self.working_dir,
+                    },
                 },
-            },
-        )
+            )
+
+            self._test(container)
 
-        with open(os.path.join(self.working_dir, 'busybox.log'), 'wb') as f:
+    def _test(self, container):
+        with open(os.path.join(self.working_dir, f'{container.name}.log'), 'wb') as f:
             f.write(b'Busybox output 1\n')
 
         proc = self.start_beat()
-        docker_client.images.pull('busybox')
-        docker_client.containers.run('busybox', 'sleep 1')
-
         self.wait_until(lambda: self.log_contains('Starting runner: input'))
-        self.wait_until(lambda: self.log_contains('Stopping runner: input'))
+        self.wait_until(lambda: self.output_has(lines=1))
 
         output = self.read_output_json()
         proc.check_kill_and_wait()
 
         # Check metadata is added
         assert output[0]['message'] == 'Busybox output 1'
-        assert output[0]['container']['image']['name'] == 'busybox'
-        assert output[0]['docker']['container']['labels'] == {}
+        assert output[0]['container']['name'] == container.name
+        assert output[0]['docker']['container']['labels'] == container.labels
         assert 'name' in output[0]['container']
 
         self.assert_fields_are_documented(output[0])
+
+    @contextmanager
+    def container_running(self, image_name='busybox:latest'):
+        import docker
```

Comment:
I guess this is required to avoid some circular import issue?

Reply:
It was already being imported from the tests, but not sure why, and I didn't want to change this. Maybe this was done to avoid importing `docker` when integration tests are disabled.